DiagonalTraversal.java

```diff
@@ -0,0 +1,63 @@
+/**
+Problem: Diagonal Traversal
+Time Complexity : O(m*n), where m is the number of rows, n is the number of columns of the given matrix.
+Space Complexity : O(1), as we are just returning the result array and not creating an additional space for the purpose of solving.
```

Comment:
That's right! There wasn't any extra/auxiliary Data Structure created for the purpose of computations!